prdoc/pr_3979.prdoc

```diff
@@ -0,0 +1,19 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Remove `para_id()` from `CoreState` in polkadot primitives
+
+doc:
+  - audience: "Node Dev"
+    description: |
+      With Coretime there is no more a static 1:1 mapping between core index and para id. This
+      mapping should be obtained from the scheduler/claimqueue on block by block basis. This change
+      removes `para_id()` from `CoreState` as it doesn't serve its purpose anymore.
+
+crates:
+  - name: polkadot-primitives
+    bump: major
```

Comment:
Not sure about this one. We are removing a pub method so it's should be a major bump?

Reply:
I think so, it is a breaking change

Reply:
Good point. Let's deprecate it first then.

Reply:
And just in case people are actually using it, let's also fix it. (Returning the scheduled para_id also on an occupied core.) This should not break any code, as up until now this was the same thing.

Reply:
@sandreim could you please have another look at the prdoc? Does the updated description make sense?